src/main/java/com/parasoft/demoapp/service/GlobalPreferencesService.java

```diff
@@ -314,6 +350,39 @@ private void handleRestEndpoints(GlobalPreferencesEntity currentPreferences,
         currentPreferences.setRestEndPoints(endpoints);
     }
 
+    private void handlePreferencesMQ(GlobalPreferencesEntity currentPreferences,
```

Comment:
I think we need to validate the properties first and then set the values for them. Like
```java
    private void refreshInventoryQueueDestinations(GlobalPreferencesEntity currentPreferences,
                    GlobalPreferencesDTO globalPreferencesDto) throws ParameterException {

        Boolean mqProxyEnabled = globalPreferencesDto.getMqProxyEnabled();
        ParameterValidator.requireNonNull(mqProxyEnabled, "Mq enabled must not be null");

        MqType mqType = globalPreferencesDto.getMqType();
        String orderServiceDestinationQueue = globalPreferencesDto.getOrderServiceDestinationQueue();
        String orderServiceReplyToQueue = globalPreferencesDto.getOrderServiceReplyToQueue();
        String inventoryServiceReplyToQueue = globalPreferencesDto.getInventoryServiceReplyToQueue();
        String inventoryServiceDestinationQueue = globalPreferencesDto.getInventoryServiceDestinationQueue();

        if(Boolean.TRUE.equals(mqProxyEnabled)) {
            ParameterValidator.requireNonNull(mqType, "Mq type must not be null");
            ParameterValidator.requireNonBlank(orderServiceDestinationQueue, GlobalPreferencesMessages.ORDER_SERVICE_DESTINATION_QUEUE_CANNOT_BE_NULL);
            ParameterValidator.requireNonBlank(orderServiceReplyToQueue, GlobalPreferencesMessages.ORDER_SERVICE_REPLY_TO_QUEUE_CANNOT_BE_NULL);
            ParameterValidator.requireNonBlank(inventoryServiceReplyToQueue, GlobalPreferencesMessages.INVENTORY_SERVICE_REPLY_TO_QUEUE_CANNOT_BE_NULL);
            ParameterValidator.requireNonBlank(inventoryServiceDestinationQueue, GlobalPreferencesMessages.INVENTORY_SERVICE_DESTINATION_QUEUE_BE_NULL);
        }

        if(mqType == null) {
            mqType = MqType.ACTIVE_MQ;
        }

        currentPreferences.setMqProxyEnabled(mqProxyEnabled);
        currentPreferences.setMqType(mqType);
        currentPreferences.setOrderServiceDestinationQueue(orderServiceDestinationQueue);
        currentPreferences.setOrderServiceReplyToQueue(orderServiceReplyToQueue);
        currentPreferences.setInventoryServiceReplyToQueue(inventoryServiceReplyToQueue);
        currentPreferences.setInventoryServiceDestinationQueue(inventoryServiceDestinationQueue);
    }
```